### DISCUSSION TITLE
Configuration settings for neovim

### DISCUSSION THREAD

**Discussion by todorovski-m** [fortran-lang/fortls#147] [Q&A] [answered]
Hi,
I was trying to configure `fortls` in neovim with partial success. I have installed `conda install -c conda-forge fortls` and in my `init.lua` I have this single line `require'lspconfig'.fortls.setup{}`
I was performing test on a small program. In the first case `fortls` immediately spotted the error in the code
![1](https://user-images.githubusercontent.com/17765294/174178258-19cffb16-3671-4aac-bcc9-75f74e4652d6.png)
But the following error was ignored
![2](https://user-images.githubusercontent.com/17765294/174178573-3530daae-a648-49e7-a613-13301c4fa623.png)
However the VS Code extension works well and detected the error
 ![3](https://user-images.githubusercontent.com/17765294/174180869-01ebdf3d-dd9f-4fd1-b93a-10dcc191b574.png)
Could you please help me to solve the problem in neovim? What I'm missing there?

**Comment by gnikit** [comment, marked as answer]
So what you see in VS Code is a combination of `fortls` diagnostics + linter diagnostics.
The diagnostics provided by `fortls` are as follows, Everything else is from `gfortran`:
- Diagnostics
  - Multiple definitions with the same variable name
  - Variable definition masks definition from parent scope
  - Missing subroutine/function arguments
  - Unknown user-defined type used in `TYPE`/`CLASS` definition
    (only if visible in project)
  - Unclosed blocks/scopes
  - Invalid scope nesting
  - Unknown modules in `USE` statement
  - Unimplemented deferred type-bound procedures
  - Use of non-imported variables/objects in interface blocks
  - Statement placement errors (`CONTAINS`, `IMPLICIT`, `IMPORT`)

**Reply by todorovski-m** [reply]
Thank you very much for your detailed answer. It helped me a lot. It seems that [ALE (Asynchronous Lint Engine)](https://github.com/dense-analysis/ale) in combination with neovim LSP does the same as your VS Code extension. Now I get diagnostics from both `fortls` and `gfortran`.

**Reply by gnikit** [reply]
Glad I was able to help and nice to see that neovim has a linter for Fortran too!
